src/BlockchainWriter/Router.ts

```diff
@@ -29,40 +29,51 @@ export class Router {
   }
 
   onBatchWriterCreateNextBatchSuccess = async (message: any): Promise<void> => {
+    const logger = this.logger.child({ method: 'onBatchWriterCreateNextBatchSuccess' })
     const messageContent = message.content.toString()
     const { fileHashes, directoryHash } = JSON.parse(messageContent)
-    if (fileHashes.length > 0)
-      this.messaging.publish(Exchange.BlockchainWriterTimestampRequest, { fileHashes, directoryHash })
+    try {
+      if (fileHashes.length > 0)
+        await this.messaging.publish(Exchange.BlockchainWriterTimestampRequest, { fileHashes, directoryHash })
+    } catch (error) {
+      logger.error('Failed to publish BlockchainWriterTimestampRequest')
+    }
   }
 
   onBlockchainWriterTimestampRequest = async (message: any): Promise<void> => {
-    const logger = this.logger.child({ method: 'onClaimIPFSHash' })
+    const logger = this.logger.child({ method: 'onBlockchainWriterTimestampRequest' })
 
     const messageContent = message.content.toString()
     const { fileHashes, directoryHash } = JSON.parse(messageContent)
 
-    logger.trace(
-      {
-        fileHashes,
-        directoryHash,
-      },
-      'Timestamping requested'
-    )
+    logger.trace('Timestamp request', {
+      fileHashes,
+      directoryHash,
+    })
 
+    try {
+      await this.createTimestampRequest({ fileHashes, directoryHash })
+      logger.trace('Timestamp request success', { fileHashes, directoryHash })
```

Comment:
Objects come before the logging message in logging calls.

Should be:

```ts
logger.trace({ fileHashes, directoryHash }, 'Timestamp request success')
```

Reply:
Are you open to reversing it like I have it? I found it makes reading the messages much easier when  the starting line of the message doesn't get pushed around by random object line widths

Reply:
PinoJS expects objects as the first argument, message as the second AFAIK. 

Just tried running your version and the object got appended to the message. 

See for example:
```
[2018-06-30T01:41:36.991Z] INFO (... on ...): Loaded Configuration and merged with defaults
    rabbitmqUrl: "amqp://localhost"
    mongodbUrl: "mongodb://localhost:27017/poet"
    ipfsUrl: "http://localhost:5001"
    insightUrl: "https://test-insight.bitpay.com/api"
    apiPort: 18080
    poetNetwork: "BARD"
    poetVersion: [
      0,
      0,
      0,
      3
    ]
    minimumBlockHeight: 1335339
    blockchainReaderIntervalInSeconds: 5
    enableTimestamping: true
    timestampIntervalInSeconds: 5
    downloadIntervalInSeconds: 5
    downloadRetryDelayInMinutes: 1
    downloadMaxAttempts: 20
    downloadTimeoutInSeconds: 30
    loggingLevel: "info"
    loggingPretty: true
    batchIntervalInSeconds: 60
    readDirectoryIntervalInSeconds: 10
    bitcoinAddress: "..."
    bitcoinAddressPrivateKey: "..."
    xforceBlockHeight: 1335603
[2018-06-30T01:41:36.994Z] INFO (... on ...): Loaded Configuration and merged with defaults {"rabbitmqUrl":"amqp://localhost","mongodbUrl":"mongodb://localhost:27017/poet","ipfsUrl":"http://localhost:5001","insightUrl":"https://test-insight.bitpay.com/api","apiPort":18080,"poetNetwork":"BARD","poetVersion":[0,0,0,3],"minimumBlockHeight":1335339,"blockchainReaderIntervalInSeconds":5,"enableTimestamping":true,"timestampIntervalInSeconds":5,"downloadIntervalInSeconds":5,"downloadRetryDelayInMinutes":1,"downloadMaxAttempts":20,"downloadTimeoutInSeconds":30,"loggingLevel":"info","loggingPretty":true,"batchIntervalInSeconds":60,"readDirectoryIntervalInSeconds":10,"bitcoinAddress":"...","bitcoinAddressPrivateKey":"...","xforceBlockHeight":1335603}
```

Disabling pretty-printing makes the problem easier to spot:

```json
{"level":30,"time":1530322806800,"msg":"Loaded Configuration and merged with defaults","pid":0,"hostname":"...","rabbitmqUrl":"amqp://localhost","mongodbUrl":"mongodb://localhost:27017/poet","ipfsUrl":"http://localhost:5001","insightUrl":"https://test-insight.bitpay.com/api","apiPort":18080,"poetNetwork":"BARD","poetVersion":[0,0,0,3],"minimumBlockHeight":1335339,"blockchainReaderIntervalInSeconds":5,"enableTimestamping":true,"timestampIntervalInSeconds":5,"downloadIntervalInSeconds":5,"downloadRetryDelayInMinutes":1,"downloadMaxAttempts":20,"downloadTimeoutInSeconds":30,"loggingLevel":"info","loggingPretty":false,"batchIntervalInSeconds":60,"readDirectoryIntervalInSeconds":10,"bitcoinAddress":"...","bitcoinAddressPrivateKey":"...","xforceBlockHeight":1335603,"v":1}
{"level":30,"time":1530322806800,"msg":"Loaded Configuration and merged with defaults {\"rabbitmqUrl\":\"amqp://localhost\",\"mongodbUrl\":\"mongodb://localhost:27017/poet\",\"ipfsUrl\":\"http://localhost:5001\",\"insightUrl\":\"https://test-insight.bitpay.com/api\",\"apiPort\":18080,\"poetNetwork\":\"BARD\",\"poetVersion\":[0,0,0,3],\"minimumBlockHeight\":1335339,\"blockchainReaderIntervalInSeconds\":5,\"enableTimestamping\":true,\"timestampIntervalInSeconds\":5,\"downloadIntervalInSeconds\":5,\"downloadRetryDelayInMinutes\":1,\"downloadMaxAttempts\":20,\"downloadTimeoutInSeconds\":30,\"loggingLevel\":\"info\",\"loggingPretty\":false,\"batchIntervalInSeconds\":60,\"readDirectoryIntervalInSeconds\":10,\"bitcoinAddress\":\"...\",\"bitcoinAddressPrivateKey\":\"...\",\"xforceBlockHeight\":1335603}","pid":0,"hostname":"...","v":1}
```

See how in the second case the passed object got stringified and appended to `msg`, whereas in the first case the passed object got merged with the rest of the log message object. 

We want the first because it allows us to filter the logs by properties we pass. We could filter by `module=BatchReader` and `level>30`, for example. It also makes extracting information out of the logs easier, since there's no room for interpretation.

The order of the arguments was shocking to me too at first, but I think I've found the logic. Function arguments can be considered more and less significant depending on their position. For example: optional parameters usuall (always?) come after mandatory ones. 

The philosophy of PinoJS, Bunyan and other structured loggers is built upon the idea of logs needing to be machine-readable above all else. The order of the arguments follows.

Reply:
Cool thanks for the index in-depth answer :D

I will update my code to the correct order.

@jwicks31 ping, just FYI this is good to know.

Reply:
and I just noticed the typscript arguments hints for these functions. Can't believe I missed that.